compiler/src/iree/compiler/Codegen/SPIRV/SPIRVVectorizeLoadStore.cpp

```diff
@@ -411,8 +411,24 @@ MemRefConversionPattern<OpTy>::getVectorizedMemRefType(
   if (newShape.back() % ratio != 0) return {};
   newShape.back() = newShape.back() / ratio;
 
-  return MemRefType::get(newShape, vectorType, MemRefLayoutAttrInterface(),
-                         type.getMemorySpace());
+  MemRefLayoutAttrInterface layout = {};
+  if (auto stridedLayout = type.getLayout().dyn_cast<StridedLayoutAttr>()) {
+    auto offset = stridedLayout.getOffset();
+    if (offset != ShapedType::kDynamic) {
+      offset = offset / ratio;
+    }
+
+    auto strides = llvm::to_vector(stridedLayout.getStrides());
```

Comment:
Same thing here re indexing logic that needs to be commonalized, tested once in a central place and reused.